splunk_add_on_ucc_framework/generators/conf_files/create_commands_conf.py

```diff
@@ -0,0 +1,51 @@
+#
+# Copyright 2025 Splunk Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+from typing import Any, Dict, Union
+
+from splunk_add_on_ucc_framework.generators.conf_files import ConfGenerator
+
+
+class CommandsConf(ConfGenerator):
+    __description__ = (
+        "Generates `commands.conf` for custom commands provided in the globalConfig."
+    )
+
+    def _set_attributes(self, **kwargs: Any) -> None:
+        self.conf_file = "commands.conf"
+        if self._global_config and self._global_config.has_custom_search_commands():
+            self.command_names = []
+            for command in kwargs["custom_search_commands"]:
+                self.command_names.append(command["commandName"])
+
+    def generate_conf(self) -> Union[Dict[str, str], None]:
```

Comment:
What happens if there is a `commands.conf` file already in the `package` folder?

Reply:
If the `commands.conf` is present in the package folder, it would overwrite the one we generated, as it is done for the other conf files (except `app.conf`). Do we want to merge the generated and source content of the conf files?